libs/model/src/policies/ContestWorker.policy.ts

```diff
@@ -1,18 +1,77 @@
 import { schemas, type Policy } from '@hicommonwealth/core';
+import { commonProtocol, models } from '@hicommonwealth/model';
 
 const inputs = {
   ThreadCreated: schemas.events.ThreadCreated,
+  ThreadUpvoted: schemas.events.ThreadUpvoted,
   CommentCreated: schemas.events.CommentCreated,
 };
 
 export const ContestWorker: Policy<typeof inputs> = () => ({
   inputs,
   body: {
     ThreadCreated: async ({ name, payload }) => {
-      console.log(name, payload.thread);
+      console.log(name, payload);
+      const { threadId, userAddress } = payload;
+
+      const contestAddress = '0x123'; // TODO: get contest address from contest metadata projection
+      const contentUrl = `https://`; // TODO: generate content URL programmatically?
+
+      const web3Client = await createWeb3Client(threadId);
+
+      const { contentId } = await commonProtocol.contestHelper.addContent(
+        web3Client,
+        contestAddress,
+        userAddress,
+        contentUrl,
+      );
+
+      // TODO: save content ID for later use in ThreadUpvoted event
```

Comment:
content ID is emitted by on-chain contest aggregates, and picked up the contests projection

Reply:
ok, so that's handled outside of this scope?